### PR TITLE
fix: convert large integers to strings

### PR DESCRIPTION
**Pull request by matthewelwell**
## Changes

Related issue: https://github.com/Flagsmith/flagsmith/issues/123

This PR is a step in the right direction, but still needs further resolution as defined in https://github.com/Flagsmith/flagsmith/issues/1667. This simply replicates the behaviour already implemented by the FE in all cases except this one (because the conversion is happening on the backend). 

## How did you test this code?

Updated an existing unit test.

### CODE REVIEW

File: api/features/utils.py [hunk 1]
```diff
@@ -8,34 +11,38 @@ def get_value_type(value: str) -> str:
...
     e.g. "12" -> "int", "12.34" -> "float", etc.
     """
-    if is_integer(value):  # type: ignore[no-untyped-call]
+    if is_32_bit_integer(value):
         return INTEGER
-    elif is_boolean(value):  # type: ignore[no-untyped-call]
+    elif is_boolean(value):
         return BOOLEAN
     else:
         return STRING
```
Comment (Zaimwa9) [inline api/features/utils.py:19]: We could use an enum for the feature value types available for stronger typing maybe with the ones in `value_types.py`
```
INTEGER = "int"
STRING = "unicode"
BOOLEAN = "bool"
FLOAT = "float"
```

Comment (matthewelwell) [reply api/features/utils.py:19]: Yeah, agreed. I kind of hate enums in python but let me see what it looks like if I replace these strings with them.

File: api/tests/unit/features/test_unit_features_utils.py [hunk 2]
```diff
@@ -8,6 +9,7 @@
     "value, expected_type",
     (
         ("1", INTEGER),
+        (f"{MAX_32_BIT_INTEGER + 1}", STRING),
```
Comment (Zaimwa9) [inline api/tests/unit/features/test_unit_features_utils.py:12]: Maybe super edge case but are we safe with the negative limit (-MAX_32_BIT_INTEGER) ?
I ran the tests with 
```
        (f"{-MAX_32_BIT_INTEGER - 100}", STRING),
```
It fails (returning integer) and I wonder if it would break ?

**Review by Zaimwa9 — CHANGES_REQUESTED** [review]
One question to address, once resolved (needed or not) looks good to go

File: api/tests/unit/features/test_unit_features_utils.py (hunk 2 above)
Comment (matthewelwell) [reply api/tests/unit/features/test_unit_features_utils.py:12]: Good catch! Fixed [here](https://github.com/Flagsmith/flagsmith/pull/5314/commits/7595c67a3a93ff164e426deae23167f5cd454644), and test updated.

File: api/features/utils.py (hunk 1 above)
Comment (matthewelwell) [reply api/features/utils.py:19]: This turned into a pretty large refactor that I'm not particularly keen on including in this PR (see [here](https://github.com/Flagsmith/flagsmith/pull/5314/commits/8193baa423fcc8aa14a9681bd16132656cc604b2)). It might be something to add in a separate PR, however.

Comment (Zaimwa9) [reply api/features/utils.py:19]: Alright, just to confirm, do you need to revert the commit with enum or can I test the PR with it included ?

Comment (matthewelwell) [reply api/features/utils.py:19]: Yeah, I'll revert that commit, and probably submit a separate PR with just that commit in it.

Comment (matthewelwell) [reply api/features/utils.py:19]: I mostly just pushed it here as a bit of a 'show and tell' 🙂

**Review by Zaimwa9 — APPROVED** [review]
Thanks for the changes

**Comment by matthewelwell** [comment]
Note:

I have manually tested the following scenarios in staging and validated that it works as expected: 

 1. Create a feature with a numeric value larger than the max size and verify that it is created successfully, and the value is cast to a string
 2. Create a feature with a numeric value smaller than the max size and verify that it is created successfully, and the value is not cast to a string (it is an integer)
 3. Create a new environment in a project that has a feature with an initial value which is numeric and larger than the max size, and verify that the new environment is correctly created, with the feature state, and the feature state's value is cast to a string
